File: configmodels/model_subs_data.go
// SPDX-FileCopyrightText: 2021 Open Networking Foundation <info@opennetworking.org>
//
// SPDX-License-Identifier: Apache-2.0
//

package configmodels

import (
	"github.com/omec-project/openapi/models"
)

type SubsData struct {
	PlmnID                            string                                     `json:"plmnID"`
	UeId                              string                                     `json:"ueId"`
	AuthenticationSubscription        models.AuthenticationSubscription          `json:"AuthenticationSubscription"`
	AccessAndMobilitySubscriptionData models.AccessAndMobilitySubscriptionData   `json:"AccessAndMobilitySubscriptionData"`
	SessionManagementSubscriptionData []models.SessionManagementSubscriptionData `json:"SessionManagementSubscriptionData"`
	SmfSelectionSubscriptionData      models.SmfSelectionSubscriptionData        `json:"SmfSelectionSubscriptionData"`
	AmPolicyData                      models.AmPolicyData                        `json:"AmPolicyData"`
	SmPolicyData                      models.SmPolicyData                        `json:"SmPolicyData"`
	FlowRules                         []FlowRule                                 `json:"FlowRules"`
}

type SubsOverrideData struct {
	PlmnID         string `json:"plmnID"`
	OPc            string `json:"opc"`
	Key            string `json:"key"`
	SequenceNumber string `json:"sequenceNumber"`
}

File: ui/frontend_files/modules/objectsModels/model_subs_data.js
/**
 * JavaScript equivalent of Go structs from model_subs_data.go
 */

import { FlowRule } from "./model_flow_rule";

// TODO: implement models
// AuthenticationSubscription        models.AuthenticationSubscription          `json:"AuthenticationSubscription"`
// AccessAndMobilitySubscriptionData models.AccessAndMobilitySubscriptionData   `json:"AccessAndMobilitySubscriptionData"`
// SessionManagementSubscriptionData []models.SessionManagementSubscriptionData `json:"SessionManagementSubscriptionData"`
// SmfSelectionSubscriptionData      models.SmfSelectionSubscriptionData        `json:"SmfSelectionSubscriptionData"`
// AmPolicyData                      models.AmPolicyData                        `json:"AmPolicyData"`
// SmPolicyData                      models.SmPolicyData  
export const SubsData = () => {
  return {
    plmnID: "",
    ueId: "",
    AuthenticationSubscription: {},
    AccessAndMobilitySubscriptionData: {},
    SessionManagementSubscriptionData: [],
    SmfSelectionSubscriptionData: {},
    AmPolicyData: {},
    SmPolicyData: {},
    FlowRules: [] // FlowRule|null
  };
};

export const SubsOverrideData = () => {
  return {
    "plmnID": "",
    "opc": "",
    "key": "",
    "sequenceNumber": "",
    "k4_sno": null,
    "encryptionAlgorithm": null
  };
};
